README.md

```diff
@@ -10,15 +10,42 @@ Go clients for talking to a [kubernetes](http://kubernetes.io/) cluster.
 * The `transport` package is used to set up auth and start a connection.
 * The `tools/cache` package is useful for writing controllers.
 
-### Releases
+### Versioning
 
-Each top-level folder (e.g., 1.4) contains a release of clients and their dependencies.
+`client-go` will follow [semver](http://semver.org/). We will not make backwards-incompatible changes without incrementing the major version number. A branch will be created for each increase in the major version number (first digit of the version) or minor version number (second digit of the version). A new tag will be created for each increase in the patch version number (third digit of the version).
 
-client-go has the same release cycle as the Kubernetes main repository. For example, in the 1.4 release cycle, the contents in `1.4/` folder are subjected to changes. Once 1.4 is released, new changes will go into the `1.5/` folder. We will make great efforts to not change the public interface of a version of the client once that version has been released. We may change the interface between versions. Old versions of the client will be retained for two release cycles.
+The master branch will track the head in the main Kubernetes repo and accumulating changes. We will bump the client-go version when:
+* A new major (very rare) or a new minor version is released in the Kubernetes main repository.
+* A feature in the master branch is popular and users want it in a stable branch. 
+
+Depending on the nature of the changes accumulated when we bump the version, we will bump version number according to semver.
+
+Note that these are the versions *of the client library itself*, not the versions of the Kubernetes clusters which the library will talk to. 
+
+#### Compatibility: Your code <-> client-go
+
+`client-go` will follow [semver](http://semver.org/), so your code keeps compatible until the major version of client-go gets bumped.
```

Comment:
We don't explicitly say so, but this guarantee only applies to **explicitly supported cluster versions**, introduced below.  I think this might be easier to understand if the first section is `client-go <-> Kubernetes` and we say each release includes the set of cluster versions it will work with.  Be sure to pick a version that is compatible with your cluster(s).

Then in the next section we'll be able to say

> [...] client-go will be backward-compatible with your code **for supported versions** [...]